### ISSUE TITLE
Train recognition: the learning rate scheduler

### ISSUE THREAD

**Issue by zeakey** [deepinsight/insightface#981]
I got confused on the learning rate scheduler in [recognition/train.py](https://github.com/deepinsight/insightface/blob/4a4b8d03fec981912fdef5b3232a37a827cbeed6/recognition/train.py#L291-L303)

Relevant code is:
```
def _batch_callback(param):
    #global global_step
    global_step[0]+=1
    mbatch = global_step[0]
    for step in lr_steps:
    if mbatch==step:
        opt.lr *= 0.1
        print('lr change to', opt.lr)
        break

    _cb(param)
    if mbatch%1000==0:
    print('lr-batch-epoch:',opt.lr,param.nbatch,param.epoch)
```

So what's the difference between `mbatch`, `nbatch` and `global_step` here?

For me I trained with the default configurations with batch-size=110 (the maximal bs I can have, due to the GPUs Mem limitation) on a 4-GPUs machine.

From my log the learning rate changes at a very strange batch index:
```
lr change to 0.010000000000000002
lr-batch-epoch: 0.010000000000000002 7368 7
```

Can someone explain the mechanism here？

**Comment by zeakey** [comment]
And another question is that when should the training be terminated? I cannot figure out it from the code itself because the `num_epochs` is set to 9999 and `max_steps` is set to 0.

**Comment by nttstar** [comment]
``mbatch`` is the number of batches we have processed.
You have to kill the training process manually.
